pkg/cloudevents/transport/http/transport.go

```diff
@@ -146,6 +145,16 @@ func copyHeaders(from, to http.Header) {
 	}
 }
 
+// Ensure to is a non-nil map before copying
+func copyHeadersEnsure(from http.Header, to *http.Header) {
```

Comment:
`http.Header` is a `[][]string` and by definition it is already a pointer. No need to make it a pointer pointer.

Reply:
In this case there is - it's the classic C "a writable pointer parameter is a pointer to a pointer". The zero value of a map is nil, *not* an empty map. So if you have
    var r req http.Request
    copyHeadersEnsure(m, r.Header)
you will be passing a *nil map* to copyHeadersEnsure, not an existing map that can be updated. The only way a function like copyHeadersEnsure can change the value of r.Header to be a *different map*  (not just modify an existing non-nil map) is by passing a pointer. The nil map is different from any useable map.

I find it pretty annoying that the zero value of a map in Go is not a usable map, but I can kinda see why they did it from a performance perspective. The same holds for channels and other "pointer like" types - they are pointer like including in the sense that they can be nil, and you have to program for that.